blog/articles/event-drive-app-knative-eventing-kogito.md

```diff
@@ -0,0 +1,131 @@
+
+![Kogito Runtime workflow parse process](images/kogito/kogito-process-workflow-file.png)*Kogito Runtime parsing flow*
+
+In this example, we have also added the [Knative Kogito Eventing plugin](https://docs.jboss.org/kogito/release/latest/html_single/#con-knative-eventing_kogito-developing-process-services) to the project, which means that it can accept CloudEvent objects through HTTP on the root path. For example:
+
+```bash
+$ curl -X POST \
+      -H "content-type: application/json"  \
+      -H "ce-specversion: 1.0"  \
+      -H "ce-source: /hospital/entry"  \
+      -H "ce-type: new.patients.events"  \
+      -H "ce-id: 12346"  \
+      -d "{ \"name\": \"Mick\", \"dateOfBirth\": \"1983-08-15\", \"symptoms\":[\"seizures\"]}" \
+http://localhost:8080
+```
+
+When sending this request to the service, a new instance of the workflow starts, and performs all of the operations defined in the spec.
+
+There's much more included in this example. For comprehensive instructions on how to deploy and build it, please see the example page on the [Github repository](https://github.com/kiegroup/kogito-examples/tree/master/serverless-workflow-functions-events-quarkus).
+
+### Knative Eventing Integration
+
+Based on this generated service, you can build [an image](https://docs.jboss.org/kogito/release/latest/html_single/#proc-kogito-deploying-on-kubernetes_kogito-deploying-on-openshift) to be deployed with the [Kogito Operator](https://docs.jboss.org/kogito/release/latest/html_single/#con-kogito-operator-and-cli_kogito-deploying-on-openshift) on a Kubernetes cluster with [Knative Eventing](https://knative.dev/docs/eventing/) installed. The Operator will create all the necessary Knative resources to configure this service and subscribe it to the Knative [broker](https://knative.dev/docs/eventing/broker/):
+
+![Knative and Kogito integration](images/kogito/knative-eventing-kogito-operator.png)*Knative Eventing and Kogito Operator integration*
+
+The Kogito Operator creates a Knative [Trigger](https://knative.dev/docs/eventing/triggers/) resource, that links the service and the broker together. In this example, it will filter events of type `new.patients.events`. This means that every time a new event of this type comes to the broker, it will be redirected to the Kogito service.
+
+The same concept also applies to [events produced by the workflow engine](https://github.com/serverlessworkflow/specification/blob/0.5.x/specification.md#event-definition). In this case, the operator will create a Knative [SinkBinding](https://knative.dev/docs/eventing/samples/sinkbinding/) resource, and will bind it to the Knative broker. Each time an event is produced by the service, a CloudEvent representing it will be sent to the broker.
```

Comment:
could we be explicit in what the `sinkbinding` subject is here?  If we're using yaml for this example, would it be possible to link to the yaml resource?

Reply:
Hi @lberk ! Thanks for your review. The `sinkbinding` here is created automatically by the Kogito Operator once a workflow is deployed, so we don't have the yaml source (I could create an example thou). See: 
https://docs.jboss.org/kogito/release/latest/html_single/#con-knative-eventing_kogito-developing-process-services

I think we can link this doc here?

In this particular example, we only have the `trigger` to be created since we listening to the new patient event.

Reply:
Hi! Fair enough, I think I was just trying to say 'I'm not sure what exactly the subject is' from the link you provided I think this diagram summarizes it (and if we could figure out how to communicate this somehow in the blog post, it'd be great!) https://docs.jboss.org/kogito/release/latest/html_single/kogito/bpmn/kogito-knative-impl-producing-event.png

Reply:
@lberk I pushed an update to the article incorporating this image and introducing it to this context, so readers can have a better view of the concept. I don't want to extend too much since this is an introductory article about this integration. I intend to keep writing more about these concepts in the future using this article as a reference. wdyt?